src/test/java/HandlerRouterTest.java

```diff
@@ -0,0 +1,60 @@
+import com.td.HttpServer.*;
+import com.td.Mocks.*;
+import java.util.HashMap;
+
+public class HandlerRouterTest extends junit.framework.TestCase {
+
+  MockFileIO mockFileIO;
+  HandlerRouter handlerRouter;
+  HttpRequest request;
+
+  protected void setUp() {
+    mockFileIO = new MockFileIO();
+    handlerRouter = new HandlerRouter(mockFileIO);
+    request = new HttpRequest("GET /file.txt HTTP/1.1", new HashMap<String, String>());
```

Comment:
It's a little weird that the setup creates a `GET` request that's only used by half of the tests.